scripts/spack/packages/netcdf-c/package.py

```diff
@@ -184,6 +184,11 @@ def configure_args(self):
         if hdf5_hl.satisfies('~shared'):
             libs.append(hdf5_hl.libs.link_flags)
 
+            # SERAC EDIT BEGIN
+            # WSL gcc 9 needs ldl and lz at the end of the link line
+            libs.append('-ldl -lz')
```

Comment:
I can't even find where these were coming from hdf5 before.. @becker33 do you have an idea why this changes in the newest Spack?  This worked prior to updating to a newer develop.

Reply:
So I don't forget this before I get to it.  Per @becker33, we should fix hdf5 to properly add `dl` to its libs when `~shared`.   I assume `-lz` is missing as well and should be added.

We think this was accidentally working before and the ordering changed and this issue was exposed. 

There is a fix I put into mfem for this as well that should be removed if this is fixed properly.

Reply:
I have also run into issues where liblua is missing `-ldl` on WSL.  I imagine this is probably related.